### PR TITLE
Add `ChangeTabletTags` RPC to `vtctl`, `ChangeTags` RPC to `vttablet`

### PR DESCRIPTION
**Pull request by timvaillancourt**
## Description

As discussed in https://github.com/vitessio/vitess/issues/16377, this PR implements a new `vtctld` and corresponding `vttablet` RPC to allow the tags of a tablet to be changed

- VTCtld RPC: `ChangeTabletTags`
- VTTablet RPC: `ChangeTags`

The new RPC supports both replacing and merging the existing tags. This merging happens on the tablet under a lock

#### CLI
```bash
tim@Tims-MacBook-Pro vitess % ./bin/vtctldclient ChangeTabletTags --help 2>&1 | head       
Changes the tablet tags for the specified tablet, if possible.

Tags must be specified as key=value pairs.

Usage:
  vtctldclient ChangeTabletTags <alias> <tablet-tag> [ <tablet-tag> ... ]

Flags:
  -h, --help      help for ChangeTabletTags
  -r, --replace   Replace all tablet tags with the tags provided. By default tags are merged/updated.
```

## Related Issue(s)

Resolves: https://github.com/vitessio/vitess/issues/16873

Original discussion: https://github.com/vitessio/vitess/issues/16377

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [x] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on CI?
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by timvaillancourt** [comment]
I assume this change would need some website docs that I'm happy to add. Any suggestions "where" this should go is appreciated!

**Comment by timvaillancourt** [comment]
👋 I've tested this on our infrastructure and things are looking good so far:

```bash
tvaillancourt@REDACTED:~$ vtctldclient $FLAGS GetTablet us_east_1b-0170348168 | jq .tags
{
  "hello": "world"
}
tvaillancourt@REDACTED:~$ vtctldclient $FLAGS ChangeTabletTags us_east_1b-0170348168 test=123
- [hello: "world"]
+ [hello: "world" test: "123"]
tvaillancourt@REDACTED:~$ vtctldclient $FLAGS GetTablet us_east_1b-0170348168 | jq .tags
{
  "hello": "world",
  "test": "123"
}
```
And `--replace` mode:
```bash
tvaillancourt@REDACTED:~$ vtctldclient $FLAGS ChangeTabletTags us_east_1b-0170348168 replace=true --replace
- [hello: "world" test: "123"]
+ [replace: "true"]
tvaillancourt@REDACTED:~$ vtctldclient $FLAGS GetTablet us_east_1b-0170348168 | jq .tags
{
  "replace": "true"
}```

File: go/vt/topotools/tablet.go [hunk 1]
```diff
@@ -59,6 +60,45 @@ func ConfigureTabletHook(hk *hook.Hook, tabletAlias *topodatapb.TabletAlias) {
...
+			return topo.NewError(topo.NoUpdateNeeded, topoproto.TabletAliasString(tabletAlias))
+		}
+		if replace || tablet.Tags == nil {
+			tablet.Tags = tabletTags
+		} else {
+			var doUpdate bool
+			for key, val := range tabletTags {
+				if val == "" {
+					delete(tablet.Tags, key)
+					doUpdate = true
```
Comment (GuptaManan100) [inline go/vt/topotools/tablet.go:81]: nit: not super important, but it is possible the key doesn't exist in tablet.Tags either, so the updation might not be required.

File: go/vt/topotools/tablet.go [hunk 2]
```diff
@@ -59,6 +60,45 @@ func ConfigureTabletHook(hk *hook.Hook, tabletAlias *topodatapb.TabletAlias) {
 	hk.ExtraEnv["TABLET_ALIAS"] = topoproto.TabletAliasString(tabletAlias)
 }
 
+// ChangeTags changes the tags of the tablet. Make this external, since these
+// transitions need to be forced from time to time.
+//
+// If successful, the updated tablet record is returned.
+func ChangeTags(ctx context.Context, ts *topo.Server, tabletAlias *topodatapb.TabletAlias, tabletTags map[string]string, replace bool) (*topodatapb.Tablet, error) {
```
Comment (GuptaManan100) [inline go/vt/topotools/tablet.go:67]: I don't see this function being used anywhere other than test code. Is this right?

**Review by GuptaManan100 — APPROVED** [review]
Rest LGTM!

File: go/vt/topotools/tablet.go (hunk 2 above)
Comment (timvaillancourt) [reply go/vt/topotools/tablet.go:67]: @GuptaManan100 correct, which is a bit confusing - but I copied `ChangeType` quite literally

File: go/vt/topotools/tablet.go (hunk 1 above)
Comment (timvaillancourt) [reply go/vt/topotools/tablet.go]: @GuptaManan100 good point! I believe I've resolved this now

File: changelog/21.0/21.0.0/summary.md [hunk 3]
```diff
@@ -218,3 +219,22 @@ work automatically during the [`MoveTables`](https://vitess.io/docs/reference/vr
 [`--remove-sharded-auto-increment` boolean flag](https://vitess.io/docs/20.0/reference/programs/vtctldclient/vtctldclient_movetables/vtctldclient_movetables_create/) and you should begin using the new
 [`--sharded-auto-increment-handling` flag](https://vitess.io/docs/21.0/reference/programs/vtctldclient/vtctldclient_movetables/vtctldclient_movetables_create/) instead. Please see the new
 [`MoveTables` Auto Increment Handling](https://vitess.io/docs/21.0/reference/vreplication/movetables/#auto-increment-handling) documentation for additional details.
+
```
Comment (rohit-nayak-ps) [inline changelog/21.0/21.0.0/summary.md:222]: We generally keep the summary short and expect users to look at the vitess.io docs for the details.

Comment (timvaillancourt) [reply changelog/21.0/21.0.0/summary.md]: @rohit-nayak-ps makes sense, updated!

**Review by rohit-nayak-ps — APPROVED** [review]

**Pull request merged**